--- /usr/share/perl5/PVE/LXC.pm.bak	2024-12-27 16:21:33.112118738 +0800
+++ /usr/share/perl5/PVE/LXC.pm	2024-12-27 16:21:33.115118739 +0800
@@ -628,6 +628,7 @@
     }
 
     my ($lxc_major, $lxc_minor) = get_lxc_version();
+    my $ostype = $conf->{ostype} || die "missing 'ostype' - internal error";
 
     my $raw = '';
 
@@ -637,15 +638,18 @@
 	# cgroups. Our pve-container@.service now starts lxc-start with `-F`,
 	# so we also don't need to worry about the new monitor cgroup to
 	# confuse systemd.
-	$raw .= "lxc.cgroup.relative = 0\n";
-
-	# To make things easier, let's keep our previous cgroup layout and
-	# simply move the monitor outside:
-	$raw .= "lxc.cgroup.dir.monitor = lxc.monitor/$vmid\n";
-	# cgroup namespace separation for stronger limits:
-	$raw .= "lxc.cgroup.dir.container = lxc/$vmid\n";
-	$raw .= "lxc.cgroup.dir.container.inner = ns\n";
-    }
+	# Patch for support oci
+    if ($ostype ne "oci") {
+		$raw .= "lxc.cgroup.relative = 0\n";
+
+		# To make things easier, let's keep our previous cgroup layout and
+		# simply move the monitor outside:
+		$raw .= "lxc.cgroup.dir.monitor = lxc.monitor/$vmid\n";
+		# cgroup namespace separation for stronger limits:
+		$raw .= "lxc.cgroup.dir.container = lxc/$vmid\n";
+		$raw .= "lxc.cgroup.dir.container.inner = ns\n";
+		}
+	}
 
     die "missing 'arch' - internal error" if !$conf->{arch};
     $raw .= "lxc.arch = $conf->{arch}\n";
@@ -653,8 +657,6 @@
     my $custom_idmap = PVE::LXC::Config->has_lxc_entry($conf, 'lxc.idmap');
     my $unprivileged = $conf->{unprivileged} || $custom_idmap;
 
-    my $ostype = $conf->{ostype} || die "missing 'ostype' - internal error";
-
     File::Path::mkpath($dir);
 
     my $cfgpath = '/usr/share/lxc/config';
@@ -703,7 +705,10 @@
     # When this is enabled mounts will still remain in the monitor's namespace
     # after the container unmounted them and thus will not detach from their
     # files while the container is running!
-    $raw .= "lxc.monitor.unshare = 1\n";
+	# Patch for support oci
+	if ($ostype ne "oci") {
+	    $raw .= "lxc.monitor.unshare = 1\n";
+	}
 
     my ($cgv1, $cgv2) = PVE::CGroup::get_cgroup_controllers();
 
@@ -721,10 +726,11 @@
 	    $raw .= "lxc.cgroup2.devices.deny = c 5:1 rwm\n";
 	}
     }
-
+	# Patch for support oci
+	if ($ostype ne "oci") {
     my $ttycount = PVE::LXC::Config->get_tty_count($conf);
     $raw .= "lxc.tty.max = $ttycount\n";
-
+	}
     # some init scripts expect a linux terminal (turnkey).
     $raw .= "lxc.environment = TERM=linux\n";
 
@@ -812,6 +818,10 @@
 	foreach my $entry (@$lxcconf) {
 	    my ($k, $v) = @$entry;
 	    $had_cpuset = 1 if $k eq 'lxc.cgroup.cpuset.cpus' || $k eq 'lxc.cgroup2.cpuset.cpus';
+		# Patch for support oci/redroid
+		if ($k eq 'lxc.mount.entry' && $v =~ /\$rootfs/) {
+		$v =~ s/\$rootfs/\/var\/lib\/lxc\/$vmid\/rootfs/g;
+		}
 	    $raw .= "$k = $v\n";
 	}
     }
