--- /usr/share/perl5/PVE/LXC.pm.bak	2024-12-27 23:24:26.016310366 +0800
+++ /usr/share/perl5/PVE/LXC.pm	2025-01-18 22:47:28.813585499 +0800
@@ -1141,10 +1141,22 @@
     my $res;
     for my $interface ($config->@*) {
 	my $obj = { name => $interface->{ifname} };
+	my $list = [];
 	for my $ip ($interface->{addr_info}->@*) {
 	    $obj->{$ip->{family}} = $ip->{local} . "/" . $ip->{prefixlen};
+
+	    push(
+		@$list, {
+		    'ip-address-type' => $ip->{family},
+		    'ip-address'      => $ip->{local},
+		    'prefix'          => $ip->{prefixlen}
+		}
+	    );
 	}
+	$obj->{'ip-addresses'} = $list;
+	$obj->{'hardware-address'} = $interface->{address};
 	$obj->{hwaddr} = $interface->{address};
+
 	push @$res, $obj
     }
 
